api/src/DataPersister/ActivityDataPersister.php

```diff
@@ -2,44 +2,38 @@
 
 namespace App\DataPersister;
 
-use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use App\DataPersister\Util\AbstractDataPersister;
+use App\DataPersister\Util\DataPersisterObservable;
 use App\Entity\Activity;
+use App\Entity\BaseEntity;
 use App\Entity\ContentNode\ColumnLayout;
-use App\Entity\ContentType;
-use Doctrine\ORM\EntityManagerInterface;
 
-class ActivityDataPersister implements ContextAwareDataPersisterInterface {
-    public function __construct(private ContextAwareDataPersisterInterface $dataPersister, private EntityManagerInterface $entityManager) {
-    }
-
-    public function supports($data, array $context = []): bool {
-        return ($data instanceof Activity) && $this->dataPersister->supports($data, $context);
+class ActivityDataPersister extends AbstractDataPersister {
+    public function __construct(
+        DataPersisterObservable $dataPersisterObservable
+    ) {
+        parent::__construct(
+            Activity::class,
+            $dataPersisterObservable,
+        );
     }
 
     /**
      * @param Activity $data
-     *
-     * @return object|void
      */
-    public function persist($data, array $context = []) {
+    public function beforeCreate($data): BaseEntity {
         $data->camp = $data->category->camp;
 
-        if ('post' === ($context['collection_operation_name'] ?? null)) {
-            // TODO implement actual prototype cloning and strategy classes, this is just a dummy implementation to
-            //      fill the non-nullable field for Doctrine
-            $rootContentNode = new ColumnLayout();
-            $rootContentNode->contentType = $this->entityManager
-                ->getRepository(ContentType::class)
-                ->findOneBy(['name' => 'ColumnLayout'])
-            ;
-
-            $data->setRootContentNode($rootContentNode);
+        if (!isset($data->category?->rootContentNode)) {
+            throw new \UnexpectedValueException('Property rootContentNode of provided category is null. Object of type '.ContentNode::class.' expected.');
         }
 
-        return $this->dataPersister->persist($data, $context);
-    }
+        $rootContentNode = new ColumnLayout();
```

Comment:
Konsequenterweise könnte man diese Logik auch in der Entity machen. Findest du es besser hier?

Reply:
Ich glaube ich würde das eher hier lassen.  Die Absicht ist ja, nur so wenig Business Logic in die Entity zu drücken wie notwendig.